tests/playwright-test/only-changed.spec.ts

```diff
@@ -0,0 +1,267 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { test as baseTest, expect, playwrightCtConfigText } from './playwright-test-fixtures';
+import { execSync } from 'node:child_process';
+
+const test = baseTest.extend({
+  setupRepository: async ({ writeFiles }, use, testInfo) => {
+    const baseDir = testInfo.outputPath();
+
+    const git = (command: string) => execSync(`git ${command}`, { cwd: baseDir });
+
+    await use(async () => {
+      await writeFiles({
+        'a.spec.ts': `
+        import { test, expect } from '@playwright/test';
+        import { answer, question } from './utils';
+        test('fails', () => { expect(question).toBe(answer); });
+      `,
+        'b.spec.ts': `
+        import { test, expect } from '@playwright/test';
+        import { answer, question } from './utils';
+        test('fails', () => { expect(question).toBe(answer); });
+      `,
+        'utils.ts': `
+        export * from './answer';
+        export * from './question';
+      `,
+        'answer.ts': `
+        export const answer = 42;
+      `,
+        'question.ts': `
+        export const question = "???";
+      `,
+      });
+      git(`init --initial-branch=main`);
+      git(`config --local user.name "Robert Botman"`);
+      git(`config --local user.email "botty@mcbotface.com"`);
+      git(`add .`);
+      git(`commit -m init`);
+      return git;
+    });
+  },
+});
+
+test.slow();
+
+test('should detect untracked files', async ({ runInlineTest, setupRepository }) => {
+  await setupRepository();
```

Comment:
It seems like every test starts with `setupRepository`. You might as well make a `git` fixture that will setup the repository unconditionally. This way, any test using the `git` fixture will get the repo automatically.

Reply:
nice, good idea! done in https://github.com/microsoft/playwright/pull/31727/commits/b6ef01324fa3dd0dc6bf70d348bb38d409ce7c55